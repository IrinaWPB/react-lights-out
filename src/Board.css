.Board {
  margin: 0 auto;
}

.Board-table {
  margin: 0 auto;
}