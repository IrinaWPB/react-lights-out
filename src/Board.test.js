import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Board from './Board'

// test('renders without crashing', function() {
//     render(<Board />)
// })

it('matches snapshot', function() {
    const { asFragment } = render(<Board chanceLightStartsOn={0}/>)
    expect(asFragment()).toMatchSnapshot()
})
// 
it('matches snapshot', function() {
    const { asFragment } = render(<Board chanceLightStartsOn={1}/>)
    expect(asFragment()).toMatchSnapshot()
})

test('shows "you WON when all lights are lit', function () {
    const { getByText } = render(<Board chanceLightStartsOn={1}/>)
    const youWon = getByText('You WON!')
    expect(youWon).toBeInTheDocument()
})

test('flips correct cells', function() {
    const { getByTestId} = render(<Board chanceLightStartsOn={0}/>)
    const cell00 = getByTestId('0-0')
    const cell01 = getByTestId('0-1')
    const cell02 = getByTestId('0-2')
    const cell10 = getByTestId('1-0')
    const cell20 = getByTestId('2-0')
    const cell33 = getByTestId('3-3')
    expect(cell00).not.toHaveClass('Cell-lit')
    expect(cell01).not.toHaveClass('Cell-lit')
    expect(cell02).not.toHaveClass('Cell-lit')
    expect(cell10).not.toHaveClass('Cell-lit')
    expect(cell20).not.toHaveClass('Cell-lit')
    expect(cell33).not.toHaveClass('Cell-lit')

    fireEvent.click(cell00)
    expect(cell00).toHaveClass('Cell-lit')
    expect(cell01).toHaveClass('Cell-lit')
    expect(cell02).toHaveClass('Cell-lit')
    expect(cell10).toHaveClass('Cell-lit')
    expect(cell20).toHaveClass('Cell-lit')
    //cell that is not in the same row ar col should stay the same
    expect(cell33).not.toHaveClass('Cell-lit')
})

