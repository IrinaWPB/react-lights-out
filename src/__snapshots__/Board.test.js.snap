// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Board"
  >
    <h1
      class="Board-header"
    >
      Lights Out
    </h1>
    <table
      class="Board-table"
    >
      <tbody>
        <tr>
          <td
            class="Cell "
            data-testid="0-0"
          />
          <td
            class="Cell "
            data-testid="0-1"
          />
          <td
            class="Cell "
            data-testid="0-2"
          />
          <td
            class="Cell "
            data-testid="0-3"
          />
        </tr>
        <tr>
          <td
            class="Cell "
            data-testid="1-0"
          />
          <td
            class="Cell "
            data-testid="1-1"
          />
          <td
            class="Cell "
            data-testid="1-2"
          />
          <td
            class="Cell "
            data-testid="1-3"
          />
        </tr>
        <tr>
          <td
            class="Cell "
            data-testid="2-0"
          />
          <td
            class="Cell "
            data-testid="2-1"
          />
          <td
            class="Cell "
            data-testid="2-2"
          />
          <td
            class="Cell "
            data-testid="2-3"
          />
        </tr>
        <tr>
          <td
            class="Cell "
            data-testid="3-0"
          />
          <td
            class="Cell "
            data-testid="3-1"
          />
          <td
            class="Cell "
            data-testid="3-2"
          />
          <td
            class="Cell "
            data-testid="3-3"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="Board"
  >
    <h1
      class="Board-header"
    >
      Lights Out
    </h1>
    <div>
      You WON!
    </div>
  </div>
</DocumentFragment>
`;
